modules/apps/29-fee/ibc_module.go

```diff
@@ -0,0 +1,167 @@
+package fee
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/keeper"
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/modules/core/exported"
+)
+
+// IBCModule implements the ICS26 callbacks for the fee middleware given the fee keeper and the underlying application.
+type IBCModule struct {
+	keeper keeper.Keeper
+	app    porttypes.IBCModule
+}
+
+// NewIBCModule creates a new IBCModule given the keeper and underlying application
+func NewIBCModule(k keeper.Keeper, app porttypes.IBCModule) IBCModule {
+	return IBCModule{
+		keeper: k,
+		app:    app,
+	}
+}
+
+// OnChanOpenInit implements the IBCModule interface
+func (im IBCModule) OnChanOpenInit(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID string,
+	channelID string,
+	chanCap *capabilitytypes.Capability,
+	counterparty channeltypes.Counterparty,
+	version string,
+) error {
+	feeVersion, appVersion := channeltypes.SplitChannelVersion(version)
+	if feeVersion != "" {
```

Comment:
If FeeVersion is blank, we revert to default transfer behaviour.

i.e. fee middleware is a no-op middleware in this case